linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -400,11 +408,17 @@ MOCKABLE_FUNCTION(, THREADPOOL_WORK_ITEM_HANDLE, threadpool_create_work_item, TH
 
 **S_R_S_THREADPOOL_LINUX_05_002: [** If `work_function` is `NULL`, `threadpool_create_work_item` shall fail and return a `NULL` value. **]**
 
-**S_R_S_THREADPOOL_LINUX_05_003: [** `threadpool_create_work_item` shall allocate memory for `threadpool_work_item` of type `THREADPOOL_WORK_ITEM_HANDLE`. **]**
+**S_R_S_THREADPOOL_LINUX_05_003: [** `threadpool_create_work_item` shall call `sm_exec_begin`. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_004: [** If `sm_exec_begin` returns `SM_EXEC_REFUSED`, `threadpool_create_work_item` shall fail and return a non-zero value. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_005: [** `threadpool_create_work_item` shall allocate memory for `threadpool_work_item` of type `THREADPOOL_WORK_ITEM_HANDLE`. **]**
+
+**S_R_S_THREADPOOL_LINUX_05_006: [** If the `malloc` fails and `threadpool_work_item` is `NULL` then log the error and return `NULL`. **]**
```

Comment:
>`malloc` fails and `threadpool_work_item` is `NULL` [](http://example.com/codeflow?start=44&length=51)

This should just say 

If the malloc fails, threadpool_create_work_item shall return NULL.

because if malloc fails then there is no way for threadpool_work_item to not be NULL

Reply:
Okay, wanted to specify what variable we are malloc'ing, putting the following:

If during the initialization of `threadpool_work_item`, `malloc` fails then log the error and return `NULL`.

Reply:
The "If during the initialization of threadpool_work_item" clause seems confusing. I agree with Jelani, it should simply say "If the malloc fails, threadpool_create_work_item shall return NULL."